### PR TITLE
Some general cleanup

### LINKED ISSUE #89: Update CHANGELOG and release docs post-v2.3.0 release
**What would you like to be added**:

- Add changes to CHANGELOG.md noted from v2.3.0 release
- Add clarification to release docs based on feedback form the v2.3.0 release

**Why is this needed**:

We made some last minute updates to the changelog for the v2.3.0 release and need to make sure those changes make their way back into the  actual CHANGELOG.md. file.

Also noted some gaps in the release documentation that need to be added.

### LINKED ISSUE #90: Add ilrudie to Code Owners file
**What would you like to be added**:

- Add @ilrudie  to the `./.github/CODEOWNERS` file

**Why is this needed**:

@ilrudie has been added as a maintainer for the MagTape project

### LINKED ISSUE #91: Update PR Template
**What would you like to be added**:

- Need to add `/kind release` to PR type section in PR template

**Why is this needed**:

Additional relevance for PR template that should make the contribex a bit better.

### PR DESCRIPTION
**Pull request by phenixblue**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind bug

/kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind feature

**What this PR does / why we need it**:

- Updates the CODEOWNERS file with the new addition of @ilrudie as a MagTape Maintainer 🎉 
- Backports some small changes from the v2.3.0 release notes to CHANGELOG.md for accuracy
- Add's the `/kind release` option in the PR template
- Updates the release doc with findings from the v2.3.0 release

**Which issue(s) this PR fixes**:

Fixes #89 
Fixes #90
Fixes #91

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., usage docs, etc.**:

```docs

```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -91,6 +91,8 @@ The README has [an example](/README.md#User-defined-alert-target) of how you can
 - Bump the engineerd/setup-kind Action to v05.0 to support the [deprecations noted here](https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/) (#77)
 - Change add-path commands in rego-checks CI jobs (#77)
 - Add ci-bootstrap Make target to pin versions for specific utilities (ie. kubectl) for more consistent CI (#77)
+- Enabled descriptive names for functional tests (#87)
```
Comment (ilrudie) [inline CHANGELOG.md:94]: #86 is the right reference for the functional test enhancement.

**Review by ilrudie — COMMENTED** [review]
Looks solid. Just a few corrections/suggestions.

File: docs/release.md [hunk 2]
```diff
@@ -17,56 +17,64 @@ CHANGELOG.md snippet.
...
 1. Update the demo install reference in the README (only for stable releases)
 
-	```
-	kubectl apply -f https://raw.githubusercontent.com/tmobile/magtape/<TAG>/deploy/install.yaml
+	```shell
+	$ kubectl apply -f https://raw.githubusercontent.com/tmobile/magtape/<TAG>/deploy/install.yaml
 	```
 
-1. Commit the changes and push to remote repository.
+	NOTE: The stateof the master branch may be in flux, but we should keep this link pointed at a stable release at all times so the experience works for the average visitor to the repo.
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply docs/release.md]: Cool, changed.

File: CHANGELOG.md (hunk 1 above)
Comment (phenixblue) [reply CHANGELOG.md]: Updated.

**Review by ilrudie — APPROVED** [review]
lgtm

**Pull request merged**